src/types.ts

```diff
@@ -918,7 +921,10 @@ export class ZodString extends ZodType<string, ZodStringDef, string> {
         }
       } else if (check.kind === "url") {
         try {
-          new URL(input.data);
+          const parsed = new URL(input.data);
+          if (!urlRegex.test(parsed.host)) {
+            throw new Error();
```

Comment:
can we add an error message here and maybe a comment outlining the scenario we're guarding against?

Reply:
This would be immediately caught with built-in logic: https://github.com/ExodusMovement/zod/blob/2e8c57d910783046531973fa34de45710d18682f/src/types.ts#L926-L935

Reply:
right